File: Using Registration Model/Case 1/NotificationHubsSample.Win81-VB/SettingsHelper.vb
Imports Windows.Storage

''' <summary>
''' Define the SettingsHelper.
''' </summary>
Public NotInheritable Class SettingsHelper
    Private Sub New()
    End Sub
    Private Shared ReadOnly LocalSettings As ApplicationDataContainer
    Private Const TagsKey As String = "Tags"
    Private Const ChannelUriKey As String = "ChannelUri"
    Private Const RegistrationIdKey As String = "RegistrationId"

    ''' <summary>
    ''' Initializes static members of the <see cref="SettingsHelper"/> class.
    ''' </summary>
    Shared Sub New()
        LocalSettings = ApplicationData.Current.LocalSettings
    End Sub

    ''' <summary>
    ''' Gets or sets the tags.
    ''' </summary>
    ''' <value>The tags.</value>
    Public Shared Property Tags() As List(Of String)
        Get
            If LocalSettings.Values.ContainsKey(TagsKey) Then
                Dim tagsSaved = LocalSettings.Values(TagsKey).ToString()
                Return tagsSaved.Split(New Char() {","c}).ToList()
            End If
            Return New List(Of String)()
        End Get
        Set(value As List(Of String))
            If value Is Nothing OrElse value.Count = 0 Then
                LocalSettings.Values.Remove(TagsKey)
            Else
                LocalSettings.Values(TagsKey) = String.Join(",", value.ToArray())
            End If
        End Set
    End Property
End Class

File: Using Registration Model/Case 1/NotificationHubsSample.WPSL/SettingsHelper.cs
using System.Collections.Generic;
using System.Linq;
using Windows.Storage;

namespace NotificationHubsSample
{
    /// <summary>
    /// Define the SettingsHelper.
    /// </summary>
    public static class SettingsHelper
    {
        private static readonly ApplicationDataContainer LocalSettings;
        private const string TagsKey = "Tags";

        /// <summary>
        /// Initializes static members of the <see cref="SettingsHelper"/> class.
        /// </summary>
        static SettingsHelper()
        {
            LocalSettings = ApplicationData.Current.LocalSettings;
        }

        /// <summary>
        /// Gets or sets the tags.
        /// </summary>
        /// <value>The tags.</value>
        public static List<string> Tags
        {
            get
            {
                if (LocalSettings.Values.ContainsKey(TagsKey))
                {
                    var tags = LocalSettings.Values[TagsKey].ToString();
                    return tags.Split(',').ToList();
                }
                return new List<string>();
            }
            set
            {
                if (value == null || value.Count == 0)
                {
                    LocalSettings.Values.Remove(TagsKey);
                }
                else
                {
                    LocalSettings.Values[TagsKey] = string.Join(",", value.ToArray());
                }
            }
        }
    }
}
